README.md

```diff
@@ -115,6 +132,74 @@ following resources:
 Once you have a Google API key, copy and paste it into `web/clientId.txt`. A running
 CodeWorld instance will immediately pick up changes to this file.
 
+In general, the Google authentication system will be the easiest system to
+maintain since no local password stores are required. This is the mechanism
+used by the official, live version of CodeWorld.
+
+#### <a name="local-auth"></a> Local authentication
+
+For applications in which external dependencies, such as Google accounts, are
+not acceptable, we provide a simple local authentication system:
+
+* Uses a SQLite3 database
+* Uses good security practices by storing only BCrypt hashes of passwords
+* Uses JWT-based stateless authentication
+
+This provides a local authentication system with very similar workflows to
+Google authentication (i.e. stateless client-side sessions). Currently, no
+web-based administrative interface is provided. Instead, you can use the
+`codeworld-auth` CLI tool to manage accounts.
+
+The local authentication system may be useful for situations where an
+instructor cannot reasonably expect all students to have a valid Google account
+and in which the instructor is willing to deploy a local CodeWorld stack.
+
+##### Create account database
+
+Local authentication will be enabled if a `codeworld-auth.db` file is present
+in the application's root directory. To create this database, run the following
+from the root of the Git repository:
+
+```
+build/bin/codeworld-auth init-accounts -d codeworld-auth.db
+```
+
+This will create an empty account database with no accounts.
+
+##### Create one or more user accounts
+
+Assuming you have already created an account database as described above, you
+can create a new account as follows:
+
+```
+build/bin/codeworld-auth create-account -d codeworld-auth.db johndoe Expired
+```
+
+This will create a new account with user ID `johndoe` with a randomly generated
+password. The account will be set to "Expired" which means the user will be
+prompted to enter a new password at next sign-in time.
+
+Other subcommands are provided for updating and deleting accounts etc. For
+help:
+
+```
+build/bin/codeworld-auth --help
+```
+
+##### Create a JWT secret
+
+To use local authentication, you will also need to generate a JWT secret stored
+in a file named `codeworld-auth.txt`. This is used to sign JWT tokens passed
+back and forth between the server and the browser. From the Git repository's
+root directory, run the following command:
+
+```
+build/bin/codeworld-auth generate-secret -s codeworld-auth.txt
```

Comment:
They should be separate: there are situations where it might be necessary to regenerate the secret independently of the account database, if, for example, the secret is compromised for some reason.